OpenUpSA/wazimap-ng-ui
# Server side templating for open graph metadata

## Description

Start serving the frontend using express, templating in profile-specific metadata in the page for open graph/social media.

We're trying not to give up the ability to serve using netlify for now, so that we can do QA using deploy previews for as long as possible. This introduces a bit of duplication in:

- start commands for both parcel- and express-based serving
- both express and client-side now set the page title


## Related Issue

https://wazimap.atlassian.net/browse/WNCM-425

## How is it tested automatically?

TODO

> Is the particular functionality/structure of the backend API this relies on tested?
> What unit, integration, GUI, end-to-end tests were added to test the functionality implemented/fixed in this PR?
> What other tests does it rely on so that the new tests are sufficient?

## How should a reviewer test it locally

```
yarn build
yarn start
```
and then see the title tag returned server-side in
```
curl --silent localhost:1234 | cut -c1-200
<!DOCTYPE html><html data-wf-page="61adcab3ab671b5d2ab2da33" data-wf-site="5fd05e808ba6609d12285571"><head><meta charset="utf-8"><title>Youth Explorer</title><meta content="Discover the human stories 
```
because localhost is set as one of the hostnames for youth explorer

## Screenshots

![2022-06-11_23-35](https://user-images.githubusercontent.com/235801/173206104-6cce2709-5b47-4674-a22f-b754844964cd.png)
![Screenshot_2022-06-11_23-54-27](https://user-images.githubusercontent.com/235801/173206154-1a40e4bc-5f16-438e-a8a9-87e28f077a0a.png)


## Changelog

See what's done and pending in https://wazimap.atlassian.net/browse/WNCM-425

### Added

### Updated

### Removed


## Checklist

- [x] 👩‍🎨 does the design match the [Demo](https://youthexplorer.staging-frontend.wazimap.openup.org.za/?dev-tools=true)

### Pull Request


### Code Quality


### Testing
